adhocteam/Head-Start-TTADP
# [TTAHUB-192] fix errors when updating goals

## Description of change

[Detailed background in JIRA](https://ocio-jira.acf.hhs.gov/browse/TTAHUB-192)

> The application is logging errors seemingly around updating goals:
> ```SERVICE:ACTIVITY_REPORTS - UNEXPECTED ERROR - TypeError: Cannot read property 'update' of null```

Krys helpfully noted that findOne was returning null in some circumstances. Some research indicated that we could switch to using upsert when updating objectives, and in this way we will never have a null value for the updated goal or objective. 

## How to test
Start creating a new activity report. Confirm that adding, updating, and deleting goals and objectives works. It's also helpful to save a draft after having created some goals and objectives, navigating away, and then returning to the page.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-192


## Checklists

### Every PR


- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [n/a] API Documentation updated
- [n/a] Boundary diagram updated
- [n/a] Logical Data Model updated
- [n/a] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy


### After merge/deploy
